src/extra_files.c

```diff
@@ -90,26 +90,33 @@ static int qsort_helper(const void *A, const void *B)
 	return strcmp(((struct filerecord *)A)->filename, ((struct filerecord *)B)->filename);
 }
 
-static void handle(const char *filename, bool is_dir, bool fix)
+static bool handle(const char *filename, bool is_dir, bool fix)
 {
 	char *temp;
+	bool ret = true;
 
 	if (fix) {
 		string_or_die(&temp, "%s%s", path_prefix, filename);
 		fprintf(stderr,
 			is_dir ? "REMOVING DIR %s/\n" : "REMOVING %s\n",
 			filename);
 		if (remove(temp)) {
-			fprintf(stderr, "Removing %s failed: %s", temp, strerror(errno));
+			fprintf(stderr, "Removing %s failed: %s\n", temp, strerror(errno));
+			ret = false;
 		}
 		free_string(&temp);
 	} else {
 		printf("%s%s\n", filename, is_dir ? "/" : "");
+		ret = false;
 	}
+
+	/* return true if the function deleted a file or directory, false otherwise */
+	return ret;
 }
 
 /* expect the start to end in /usr and be the absolute path to the root */
-int walk_tree(struct manifest *manifest, const char *start, bool fix, const regex_t *whitelist)
+int walk_tree(struct manifest *manifest, const char *start, bool fix, const regex_t *whitelist,
+	      int *extraneous_files, int *files_deleted, int *files_not_deleted)
```

Comment:
Rather than adding 3 int* parameters, consider adding a pointer to a struct that contains 3 integers.

Reply:
Thanks for reviewing @icarus-sparry , that was a good suggestion. I have pushed an update that includes a struct for keeping those file counts. Please let me know if you have any concern with the new implementation.